src/App.js

```diff
@@ -22,40 +34,81 @@ const generateSquares = () => {
     }
   }
 
-  return squares;
+  return squares; // array of 3 row arrays with 3 objects
 }
 
 const App = () => {
 
   const [squares, setSquares] = useState(generateSquares());
+  const [currentPlayer, setCurrentPlayer] = useState(PLAYER_1);
+  const [winner, setWinner] = useState(null);
 
-  // Wave 2
-  // You will need to create a method to change the square 
-  //   When it is clicked on.
-  //   Then pass it into the squares as a callback
+  const updateSquares = (id) => {
+    if (winner !== null) return; // stop game after winner declared
+    let newSquares = [...squares];
 
+    // find clicked Square, then update it's value
+    for (let row = 0; row < 3; row += 1) {
+      for (let col = 0; col < 3; col += 1) {
+        let selectedSquare = newSquares[row][col];
+        if (selectedSquare.id === id) {
+          if (selectedSquare.value === '') {
+            selectedSquare.value = currentPlayer;
+            newSquares[row][col] = selectedSquare;
+          }         
+        }
+      }
+    }
+    
+    setSquares(newSquares);
+    setWinner(checkForWinner(squares));
+    // switch players
+    currentPlayer === PLAYER_1 ? setCurrentPlayer(PLAYER_2) : setCurrentPlayer(PLAYER_1)
+  }
 
   const checkForWinner = () => {
-    // Complete in Wave 3
+    // get values of each square in order
+    const squareValues = [];
+    
+    squares.map(row => 
+      row.map(square => ( 
+        squareValues.push(square.value)
+        )
+      )
+    )
```

Comment:
You can simplify this by using `.flat`:

```suggestion
    const squareValues = squares.flat();
```